### ISSUE TITLE
Refactor PaperListingForm/index

### ISSUE THREAD

**Issue by seanmalbert** [bloom-housing/bloom#1916]
Nice work @emilyjablonski! I think my main comment would be that if `src/listings/PaperListingForm/index.tsx` is getting changed to this degree, it'd be a great opportunity to refactor. The file's getting pretty large (~800 lines) and feels like it needs more of an architecture. What do you think?

_Originally posted by @jaredcwhite in https://github.com/bloom-housing/bloom/pull/1850#pullrequestreview-768467928_

### QA Review Instructions
1. go to https://dev-partners-bloom.netlify.app/
2. create/edit listings

The work here updated where and how some of the logic, particularly with formatting, is called with the listing form, so we want to make sure this didn't cause any regression. It's likely that you will find issues, so we'll need to determine whether they were existing, and therefore need to be addressed in separate issues, or directly caused by this.

**Comment by seanmalbert** [comment]
@jaredcwhite and/or @emilyjablonski , can you please add your thoughts here on how you this should be handled.

**Comment by jaredcwhite** [comment]
I reviewed the file again and thought of a few suggestions:

* The types/interfaces/default values comprise around 150 lines, seems like an easy opportunity for extraction.
* The `formatFormData` function doesn't appear to have any specific React/UI/state/etc. dependencies, so should be straightforward to extract. Might even make sense to set up as a standalone class with a pipeline of transformation methods. (I'm thinking of something like what I did previously on [`AutofillCleaner`](https://github.com/bloom-housing/bloom/blob/dev/sites/public/lib/appAutofill.ts).)
* Feeling a little overwhelmed by the `ListingForm` component itself. There's 150 lines of state/effect/callback bits alone. Maybe we can extract some conceptual slices into custom hook(s)? And for the final JSX part, maybe move some more things out into small sectional components?

Let me know what you think. 🙂
